### PR TITLE
Update static page detail api to fetch by name

### PR DESCRIPTION
**Pull request by msaqib52**
@ziafazal please review, this PR updates `api_manager.courses.views.CoursesStaticTabsDetail` API to get course tab data by tab name.

### CODE REVIEW

File: lms/djangoapps/api_manager/courses/urls.py [hunk 1]
```diff
@@ -23,7 +23,8 @@
     url(r'^{0}/groups/(?P<group_id>[0-9]+)$'.format(COURSE_ID_PATTERN), courses_views.CoursesGroupsDetail.as_view()),
     url(r'^{0}/groups/*$'.format(COURSE_ID_PATTERN), courses_views.CoursesGroupsList.as_view()),
     url(r'^{0}/overview/*$'.format(COURSE_ID_PATTERN), courses_views.CoursesOverview.as_view()),
-    url(r'^{0}/static_tabs/(?P<tab_id>[a-zA-Z0-9_+\/:-]+)$'.format(COURSE_ID_PATTERN), courses_views.CoursesStaticTabsDetail.as_view()),
+    url(r'^{0}/static_tabs/(?P<tab_identifier>[a-zA-Z0-9_+\s\/:-]+)$'.format(COURSE_ID_PATTERN),
```
Comment (ziafazal) [inline lms/djangoapps/api_manager/courses/urls.py]: I think we should keep the existing url pattern and add new after after it.

File: lms/djangoapps/api_manager/courses/tests.py [hunk 2]
```diff
@@ -866,14 +866,17 @@ def test_static_tab_list_get_invalid_course(self):
         self.assertEqual(response.status_code, 404)
 
     def test_static_tab_detail_get(self):
-        test_uri = self.base_courses_uri + '/' + self.test_course_id + '/static_tabs/syllabus'
+        # get static tab by tab name
+        test_uri = self.base_courses_uri + '/' + self.test_course_id + '/static_tabs/Static Tab'
```
Comment (ziafazal) [inline lms/djangoapps/api_manager/courses/tests.py]: I think you should create new static tab with + sign in its name to test urlencode/decode is working as expected.

File: lms/djangoapps/api_manager/courses/views.py [hunk 3]
```diff
@@ -927,35 +927,35 @@ class CoursesStaticTabsDetail(SecureAPIView):
...
-          * id: The ID of the custom page.
+          * id: The url_slug of the custom page.
 
           * name: The Display Name of the custom page.
 
           * detail: The content of the custom page as HTML.
     """
 
-    def get(self, request, course_id, tab_id):
+    def get(self, request, course_id, tab_identifier):
```
Comment (ziafazal) [inline lms/djangoapps/api_manager/courses/views.py]: I think we should keep the parameter name `tab_id` as it is used in that way throughout the codebase.

**Comment by msaqib52** [comment]
@ziafazal i have updated this PR with recommended changes, please review.

File: common/lib/xmodule/xmodule/modulestore/tests/factories.py [hunk 4]
```diff
@@ -231,6 +233,7 @@ def _create(cls, target_class, **kwargs):
         parent_location = kwargs.pop('parent_location', None)
         data = kwargs.pop('data', None)
         category = kwargs.pop('category', None)
+        name = kwargs.pop('name', None)
```
Comment (ziafazal) [inline common/lib/xmodule/xmodule/modulestore/tests/factories.py]: I think you should not modify item factory instead you should create a static tab in a course in your tests like this
```
course.tabs.append(
                    CourseTab.load('static_tab', name=name, url_slug='url_slug')
                )
                store.update_item(course, user_id)
```

File: lms/djangoapps/api_manager/courses/tests.py [hunk 5]
```diff
@@ -900,6 +903,7 @@ def test_static_tab_detail_get(self):
         self.assertTrue(tab_contents is not None)
         self.assertEqual(tab_contents, self.static_tab1.data)
 
+        # now get static tab by url_slug
```
Comment (mattdrayer) [inline lms/djangoapps/api_manager/courses/tests.py]: We should break this test up into two separate tests -- one to test retrieval by tab_name and one to retrieve by url_slug

**Comment by ziafazal** [comment]
@msaqib52 LGTM :+1:

**Pull request merged**